controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -124,3 +133,204 @@ func (obj *ocsCephRbdMirrors) ensureDeleted(r *StorageClusterReconciler, sc *ocs
+	enableRbdMirrorDebugLoggingJobName := "enable-rbd-mirror-debug-logging"
+	disableRbdMirrorDebugLoggingJobName := "disable-rbd-mirror-debug-logging"
+	enableRbdMirrorDebugLoggingCommands :=
+		`echo "starting configuration of mirror logging"
+ceph config set client.rbd-mirror.a debug_ms 1
+ceph config set client.rbd-mirror.a debug_rbd 15
+ceph config set client.rbd-mirror.a debug_rbd_mirror 30
+ceph config set client.rbd-mirror.a log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set client.rbd-mirror-peer debug_ms 1
+ceph config set client.rbd-mirror-peer debug_rbd 15
+ceph config set client.rbd-mirror-peer debug_rbd_mirror 30
+ceph config set client.rbd-mirror-peer log_file /var/log/ceph/\$cluster-\$name.log
+ceph config set mgr mgr/rbd_support/log_level debug
+		echo "completed"`
+	disableRbdMirrorDebugLoggingCommands :=
+		`echo "Removing configuration of mirror logging"
+ceph config rm client.rbd-mirror.a debug_ms
+ceph config rm client.rbd-mirror.a debug_rbd
+ceph config rm client.rbd-mirror.a debug_rbd_mirror
+ceph config rm client.rbd-mirror.a log_file
+ceph config rm client.rbd-mirror-peer debug_ms
+ceph config rm client.rbd-mirror-peer debug_rbd
+ceph config rm client.rbd-mirror-peer debug_rbd_mirror
+ceph config rm client.rbd-mirror-peer log_file
+ceph config rm mgr mgr/rbd_support/log_level
+		echo "completed"`
+
+	// Mirroring is enabled but the debug logging is disabled
+	if sc.Spec.Mirroring.Enabled && !rbdMirrorDebugLoggingEnabled {
+
+		err := r.Client.Create(r.ctx, getRbdMirrorDebugLoggingJob(sc, enableRbdMirrorDebugLoggingJobName, enableRbdMirrorDebugLoggingCommands))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return reconcile.Result{}, err
+		}
+		// Check if the job has succeeded
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, types.NamespacedName{Name: enableRbdMirrorDebugLoggingJobName, Namespace: sc.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if job.Status.Succeeded == 0 {
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
+		}
+
+		r.Log.Info("Successfully enabled rbd mirror debug logging")
+		rbdMirrorDebugLoggingEnabled = true
+	}
+	// Mirroring is disabled but the debug logging is enabled
+	if !sc.Spec.Mirroring.Enabled && rbdMirrorDebugLoggingEnabled {
+		err := r.Client.Create(r.ctx, getRbdMirrorDebugLoggingJob(sc, disableRbdMirrorDebugLoggingJobName, disableRbdMirrorDebugLoggingCommands))
+		if err != nil && !errors.IsAlreadyExists(err) {
+			return reconcile.Result{}, err
+		}
+		// Check if the job has succeeded
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, types.NamespacedName{Name: disableRbdMirrorDebugLoggingJobName, Namespace: sc.Namespace}, job)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if job.Status.Succeeded == 0 {
+			return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
```

Comment:
if the job was not successful this will block the storagecluster delete workflow also?

Reply:
IMO No, The function call is at the end, after all the calls regarding storage cluster phase/ deletion are done successfully. So I don't think it will affect any of that.